Rewrite and TypeScript migration

jonbarrow opened the pull request:
Draft for now as

- This causes several regressions (basically all protocols are not implemented now)
- The UI is no longer implemented (the structure of data has changed quite a lot, and we're deciding if we want to redo the UI)

PR rewrites all the packet parsing and session handling essentially from scratch, and migrates the app to TypeScript.

The current version of the viewer is extremely hacked together and buggy. This is due to the fact that I originally wanted to just make a CLI tool here, and hacked the UI on top. Additionally at the time we did not have Hokaku so raw RMC messages were handled completely separately from other packets, despite having nearly identical logic

This rewrite builds on what we have learned since then, focussing on a UI-first application (rather than CLI with a hacked UI on top of it), and takes notes from our Go libraries on how to handle certain patterns

Data serialization now makes much more sense and has been standardized, as opposed to the current viewer which handles some serialization differently resulting in a hacky front end

### Some key changes:

- `nex-keys.txt` has been deprecated. Since some games use use different key derivation methods it's not reliable to just use the pre-hashed keys anymore. Now all game server accounts are stored in the applications `settings.json` file as an array of objects. The objects fields are
  - `platform` - The platform the account is for. Unused currently, will be used in the future to support non-NEX platforms
  - `username` - The username. On NEX this is the string representation of the `pid` value
  - `pid` - Account PID
  - `password` - Plain-text password. Used to derive the other password, including any non-NEX standard ones in the future
  - `password_hash_old` - Pre-hashed NEX password using old key derivation
  - `password_hash_new` - Pre-hashed NEX password using new key derivation
- `titles.json` has been restructured and now contains a `settings` object for stream settings
  - Some data here, such as `string_length_size` is not currently used. It is not possible to use this information at this time, as we need to know which title the packet is for BEFORE parsing it. This is not possible to do as the title is not determined until AFTER the first `SYN` packet arrives. This means Rendez-Vous connections which use non-NEX settings will fail here
    - We may need to consider bringing these to global settings, however that will break cases like Rayman Legends on Wii U where both NEX and Rendez-Vous connections will be seen
    - Alternatively we can attempt to make a WireShark-style system where these packets are still shown in the UI, and the user is able to manually mark them as being for a title. The application will then re-run the parsing with this knowledge. However we still need a way to detect those packets, as currently bad packets are thrown away
  - We may also want to consider removing `titles.json` entirely in favor of making each title it's own JavaScript class. This way a title may define more complex settings, like a custom compression algorithm, or custom checksum algorithm (such as the custom checksums seen in WATCH_DOGS)
- All primitives have been implemented as data types
- When an RMC message is serialized to JSON, return an object whose keys are the parameter names and values are the parameters
- When a data type is serialized to JSON, the following rules apply:
  - The `toJSON` function MUST return an object
  - Each object MUST have both `__displayTypeName` and `__typeName` fields
  - `__displayTypeName` is to be used when displaying the types data in the UI. Some types, such as `List`, `Map`, etc. may reference other information about themselves in this value, making it inadequate for type referencing
  - `__typeName` is the types original name, and is to be used when the program needs to check what the type of the object is
  - Types which only reference a single piece of data, such as `String`, `Bool`, etc. will set their value in the `__value` field. This can be either a primitive value (such as the value of a `Bool`) OR another data type (such as the value of a `AnyDataHolder`)
  - Types which reference multiple pieces of data (all protocol structures, `RVConnectionData`, etc.) will set these is a `__fields` object. The keys of the `__fields` object are the field names, and the values are the field values
  - Data types which are `Structure` types store the structure version number in a `__version` field
  - In order all of these fields should be serialized, when applicable, as:
    - `__version`
    - `__displayTypeName`
    - `__typeName`
    - `__value` or `__fields`
  - Parent types have not been implemented yet, however they will likely be stored in a `__parent` field ABOVE the `__version` field
- When parsing serialized data and a `__typeName` is `Map`, this must be handled in a special way. In this case the `__value` field is used, and is set to an array (much like `List`) where each element is an object. The objects have two fields, `key` and `value` each being a data type (not a primitive). This is the only case of special handling, as ANY type may be both a key or a value which doesn't play well with JavaScript (or most languages)
- The concept of a `Session` has been introduced. A `Session` simply represents a user's play session. It parses a single packet dump to look for `Connection`s
- A `Connection` is no longer identified by a string discriminator made from the address. Now it uses a combination of the address, port, and connection stream settings. This allows for titles which reuse connections with multiple streams (such as Rayman Legends) to properly split each stream into it's own connection
- Substreams are now implemented correctly
- The basis of "special protocols" has been introduced. The "special" protocols referenced in `RVConnectionData` are a separate server to the main secure server. When the StationURL is populated and the protocol list is not empty, clients will connect to ***BOTH*** the main and special stations. When clients make an RMC request, it checks the protocol ID and decides which to send it to. No known NEX titles use this feature, however it is partially checked now (though incomplete)

jonbarrow committed:
chore: begin rewrite and TS migration

jonbarrow commented:
Since this is moving to TypeScript, we may want to consider rewriting the UI in something like React or Vue. I have tagged our web developers, @hauntii and @gitlimes, because of this

jonbarrow committed:
chore: add TODO comment about merging folders in windows folder

jonbarrow committed:
fix: emit PRUDP packets in session

jonbarrow committed:
chore: add packet serializing and better type defs/checks

jonbarrow committed:
chore(types): update type imports for Packet type def

jonbarrow committed:
chore(lint): add browser globals to eslint config

jonbarrow committed:
chore: add browserify command for building

jonbarrow committed:
refactor(rmc): update serialized RMC message data

jonbarrow committed:
refactor(prudp): update serialized PRUDPPacket/PRUDPPacketV1 data

jonbarrow committed:
chore(types): add serialized packet type def

jonbarrow committed:
fix(rmc): rename _error to error on serialized RMC message data

jonbarrow committed:
fix(protocols): assert that RMC parameters exist in method handlers

jonbarrow committed:
chore(rmc): add serialized RMC message type def

jonbarrow committed:
refactor: make toJSON methods return serialized types

jonbarrow committed:
fix(nex): fix PCAPNGParser import in session.ts

jonbarrow committed:
refactor: add global settings

jonbarrow committed:
chore: remove console log in settings.ts

jonbarrow committed:
chore: update settings types

jonbarrow committed:
refactor(nex): use new account settings to lookup NEX accounts

jonbarrow committed:
chore(nex): remove debug print in connection.ts

jonbarrow committed:
fix(nex): QResult codes are uint32 not int32

jonbarrow committed:
fix(nex): fix how error responses find their request messages

jonbarrow committed:
chore(nex): add serialized Connection data

jonbarrow committed:
fix: fix "seconds" type in PCAPNG parser

jonbarrow committed:
feat: add "time" value to packets

jonbarrow committed:
fix(nex): filter DNS packets by port

jonbarrow committed:
fix(types): add "time" value to SerializedPRUDPPacket

jonbarrow committed:
feat(nex): add stack traces to serialized packets

jonbarrow committed:
refactor(nex): pass Buffer data as array values

jonbarrow committed:
refactor: add window menu and update Settings

jonbarrow committed:
fix(types): update SerializedPacket Buffer types

jonbarrow committed:
fix: revert Recent Files dialog. Caused issues

jonbarrow committed:
fix(nex): fix raw RMC packet serializing

jonbarrow committed:
feat: add the start of UI

jonbarrow committed:
refactor: condense protocol method classes into one folder

jonbarrow committed:
feat: implement all of SecureConnection

jonbarrow committed:
feat: add NintendoLoginData/AccountExtraInfo types

jonbarrow committed:
feat: implement all of NAT Traversal

jonbarrow committed:
refactor: replace RMCMessage with type import in RMC methods

jonbarrow committed:
refactor: strongly type RMCs

jonbarrow committed:
fix: fix pSourceKey value check in connection

jonbarrow committed:
chore: remove incorrect comments from SecureConnection and NAT Traversal

jonbarrow committed:
feat: implement all of NotificationEvents

jonbarrow committed:
feat: display packet JSON in UI when selected

jonbarrow committed:
feat: start implementing MatchMaking

jonbarrow committed:
feat: add basic search to packet list

jonbarrow committed:
feat: start implementing MatchMakingExt

jonbarrow committed:
feat: start implementing MatchmakeExtension

jonbarrow committed:
chore: update account settings structure in README

jonbarrow committed:
feat: add MatchmakeExtension::FindOfficialCommunity

jonbarrow committed:
feat: add FindOfficialCommunity RMC types

jonbarrow committed:
fix: add p2p packet filtering checks

jonbarrow committed:
fix: update PIA magic check

jonbarrow committed:
chore: improve PRUDP packet filtering

ashquarky committed:
feat: Add AutoMatchmakeWithSearchCriteria_Postpone

used in YKW2 and others

jonbarrow committed:
chore: update PRUDPPacketV1 code styles

jonbarrow committed:
feat: add Charles dump support and Switch packet support

jonbarrow committed:
feat: complete TicketGranting protocol

jonbarrow committed:
feat: complete MatchMakingExt protocol

jonbarrow committed:
chore: remove dangling comma in MatchMakingExt

jonbarrow committed:
chore: npm audit fix

jonbarrow committed:
feat: complete MatchMaking protocol

jonbarrow committed:
feat: complete MatchmakeExtension protocol

jonbarrow committed:
feat: complete MatchmakeReferee protocol

jonbarrow committed:
feat: complete Ranking protocol

jonbarrow committed:
chore: update method IDs in MatchmakeReferee protocol

DaniElectra commented on src/nex/protocols/ranking/types/ranking-rank-data.ts:
```diff
@@ -0,0 +1,56 @@
+	private groups = new List(new UInt8());
+	private param = new UInt64();
+	private commonData = new RVBuffer();
+
```
Here we are missing the `updateTime` field, which was added in NEX 3.6.0: https://nintendo-wiki.pretendo.network/docs/nex/protocols/ranking/#rankingrankdata-structure

We may also want to look at the original code (pre-rewrite) to ensure we don't lose documentation that may not have been ported over to the wiki

ashquarky committed:
fix: Account for NEX versions in JoinMatchmakeSessionParam

I have the strangest feeling that I made this exact commit to 
nex-protocols-go months ago

jonbarrow commented:
Thank you for the update on `JoinMatchmakeSessionParam`. Those changes are not reflected on the wiki, I'll make a note to update that tomorrow

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/PretendoNetwork/nex-viewer?pullRequest=26) <br/>All committers have signed the CLA.

jonbarrow committed:
feat: move titles to classes rather than json

having titles as classes rather than in a json file gives us the advantage of being able to set custom protocols on each title, such as protocol ID 200 which is shared between multiple games. It also lets us now define custom validation functions for packets in cases like LM2 and WATCH_DOGS which use non-standard checksum algorithms

jonbarrow committed:
chore: removed comment in NEX Title type def

DaniElectra commented:
Why are we handling NEX protocols per game now?

jonbarrow commented:
> Why are we handling NEX protocols per game now?

2 main reasons:

1. It limits the number of checks that happen when selecting a protocol. Eventually this tool should implement all the protocols, even the ones not used by NEX, which would have wasted checks
2. It lets us more cleanly handle situations like protocol 200, where multiple games have different protocols under the same ID. With the old implementation we would have to do hacks in the protocol selector to account for these cases, but if we handle them per title then we don't have to worry about conflicts like this
3. Less important but it also somewhat acts as documentation, by clearly defining what protocols a title uses rather than blindly assuming it might have them all

This will definitely be a bit more work initially to get everything populated, but once that initial hurdle is overcome it'll be a non-issue for future titles since then we're just dealing with populating 1 title at a time rather than 80

binaryoverload committed:
ci + build: add modern eslint and fix cross platform build issues

binaryoverload committed:
build: fix copying of static files

jonbarrow committed:
fix: add missing quotes to glob-exec argument in browserify script

jonbarrow committed:
chore: remove lint script from build step, lint separately

jonbarrow committed:
fix: properly add the menu bar on MacOS

jonbarrow committed:
chore: update settings path for MacOS in README

jonbarrow committed:
feat: add friends (3DS) protocol

jonbarrow committed:
feat: add friends (Wii U) protocol

jonbarrow committed:
feat: add account management protocol

jonbarrow committed:
chore: remove duplicate AccountExtraInfo type from SecureConnection

jonbarrow committed:
chore: add account management and friends protocols to NN friends title

jonbarrow committed:
fix: update extracted field names in MiiV2

jonbarrow committed:
feat: add experimental mitmproxy flows parser

jonbarrow committed:
feat: hookup mitmproxy flows parser to UI

jonbarrow committed:
feat: add Utility protocol

jonbarrow committed:
chore: update MK8D protocol list

jonbarrow committed:
chore: remove debug console.log

jonbarrow committed:
feat: allow .flow and .flows files

jonbarrow committed:
fix: enable Edit menu so MacOS keyboard shortcuts work

jonbarrow committed:
feat: add forge builder

jonbarrow committed:
chore: add .DS_Store to .gitignore

jonbarrow committed:
chore: npm audit fix

jonbarrow committed:
chore: add notes about Switch NEX account details

jonbarrow committed:
chore: add Rendez-Vous notes to README

jonbarrow committed:
chore: add support list to README

jonbarrow committed:
chore: fix typos in README

jonbarrow committed:
feat: vendor get_3ds_pid_password homebrew

jonbarrow committed:
feat: add RemoteLogDevice protocol

jonbarrow committed:
chore: run eslint fix

jonbarrow committed:
feat: add Messaging protocol

jonbarrow committed:
fix: fix Messaging protocol ID

jonbarrow committed:
feat: add MessageDelivery protocol

jonbarrow committed:
chore: remove types from MessageDelivery protocol

jonbarrow committed:
chore: update protocol support in README

jonbarrow committed:
feat: add StorageManager protocol

jonbarrow committed:
fix: update Utility protocol method imports

jonbarrow committed:
feat: port to electron-vite and Vue

jonbarrow committed:
chore(pcapng): add checks for missing SectionHeaderBlock

jonbarrow committed:
fix(pcapng): check BOM before block length in parseSectionHeaderBlock

jonbarrow committed:
feat: add protocols to Splatoon config

jonbarrow committed:
feat: add "open settings" and "open settings folder" menu options

jonbarrow committed:
chore: npm audit fix

jonbarrow committed:
chore: fix all errors reported from "npm run typecheck"

jonbarrow committed:
chore: fix all errors reported from "npm run lint"

jonbarrow committed:
chore: add sourcemaps to vite config

jonbarrow committed:
feat: add common protocols to Smash4

jonbarrow committed:
chore: improve packet decoding stack trace bubbling

jonbarrow committed:
feat: add NintendoNotificationEvents protocol

jonbarrow committed:
chore: update @pretendonetwork/java.io to v1.0.1

jonbarrow committed:
chore: npm audit fix

jonbarrow committed:
chore: swap Edit and File menu options

jonbarrow committed:
feat: add support for Charles 5.0 chlz dumps

jonbarrow committed:
chore: make linter only check src folder

jonbarrow committed:
feat: add DataStore protocol

jonbarrow committed:
feat: migrate to v2 UI

jonbarrow committed:
chore: npm audit fix

jonbarrow committed:
feat: add proxide capture parser

jonbarrow committed:
fix: add elapsed time to raw rmc packets

jonbarrow committed:
chore: rename "time" packet field to "elapsed_time"

jonbarrow committed:
fix: account for pcap microseconds when calculating elapsed time

jonbarrow committed:
feat: implement recent file selection

jonbarrow committed:
chore: add missing protocol support to README

jonbarrow committed:
feat: add title ID to raw rmc packet overview

jonbarrow committed:
feat: add stack trace view in packet overview

jonbarrow committed:
fix: set rmc method name on packet before trying to parse

if an exception was throwing during parsing then the rmc method name would not be set on the packet

jonbarrow committed:
chore: remove redundant display name check in getTypeIcon

jonbarrow committed:
fix: correct AnyDataHolder serializing and assignment

jonbarrow committed:
feat: add config for Smash Ultimate

jonbarrow committed:
feat: add PRUDPLite game detection fallback using destination address

jonbarrow committed:
feat: add elapsed time to Charles parsers

jonbarrow committed:
chore: move session.ts out of src/nex

jonbarrow committed:
chore: rename "connections" to "PRUDP connections"

jonbarrow committed:
chore: use CircleQuestionMark for unknown types in RMCField

jonbarrow committed:
feat: add NPLN support

lots of this is still really hacky, im open to changing basically everything here

jonbarrow committed:
chore: update NPLN support in README

DaniElectra committed:
fix(ticket-granting): Support pre-NEX 3.0 AuthenticationInfo

DaniElectra committed:
feat(nex/titles): Add protocols to more titles

Also fix protocols and common protocols ordering to allow overriding
with game-specific protocols.

jonbarrow committed:
chore: update @pretendonetwork/eslint-config to 0.1.4

jonbarrow committed:
chore: npm audit fix

jonbarrow committed:
fix: load only the NPLN protobufs

fixes an issue on some platforms where Google protos were being double imported

jonbarrow committed:
fix: fix all build errors

note: the build errors were caused by type errors in the UI layer, we need stronger/better types here

DaniElectra committed:
feat: begin support for P2P packets

At the moment we only parse NAT packets, but DO packets are now visible
too.

DaniElectra committed:
feat(prudp): Add isStreamTypeX methods and complete known packet types

jonbarrow committed:
chore: add PRUDP packet type 8 to types list

jonbarrow committed:
chore: use enum values in PRUDPPacket.serializeStreamType and PRUDPPacket.serializeType

jonbarrow committed:
feat: rename RMCField to SerializedField

this structure is being reused for more than just RMC now

jonbarrow committed:
feat: make UI data driven with proper message serialization

this adds a standard form of serialization we can use to more easily add support for more protocols

jonbarrow committed:
chore: remove unused files

jonbarrow committed:
fix: clear UI sections on file load and dev hot reload

jonbarrow committed:
chore: fix all linter errors

jonbarrow committed:
feat: add http headers to NPLN request/response views

DaniElectra commented on src/electron/renderer/src/components/PacketDetails.vue:
```diff
@@ -0,0 +1,121 @@
+	];
+
+	if (props.packet) {
+		if (props.packet.stack_trace) {
```
I think we should put the stack trace as the last tab like we did previously

jonbarrow replied:
I had moved it up in case we had a protocol that added many custom tabs, so that it didn't get lost, but I can move it back if you don't think that'll happen

DaniElectra replied:
I can't think of anyhing that would use more than 2 tabs. If you can give an example then I'll be fine to keep it as-is.

jonbarrow replied:
I also can't, I only moved it as a precaution. I'll move it back then 👍

DaniElectra commented on src/nex/prudp-packetv0.ts:
```diff
@@ -0,0 +1,293 @@
+		return {
+			id: -1, // * Gets set later when emitted
+			elapsed_time: this.elapsedTime ?? 0,
+			transport: 'NEX', // TODO - p2p packets should change this
```
I mean technically NEX groups both the RDV part and the Net-Z (p2p) part

jonbarrow replied:
For visual clarity in the UI it would probably be better to still separate them though, so we know immediately at a glance what's p2p and what isn't

DaniElectra replied:
We could do something like `NEX-RDV` and `NEX-P2P` then (names are suggestions, you get the point)

jonbarrow replied:
What about just using "NEX" for the standard ones we typically think about as being "NEX" and then using "NetZ" for these p2p packets? Since we know NetZ is already under NEX

DaniElectra replied:
Not against that but I'd prefer the other way as it encourages correctness

jonbarrow replied:
I'm not sure what you mean? NetZ is correct, no? The main point of that field is to use the name of the system transporting the packets, which in this case would be NetZ? NEX is the library that handles NetZ, and it reuses PRUDP for it, but the specific system at play is still NetZ. My main concern with something like `NEX-P2P` is that it's not immediate at a glance what kind of p2p system it is (I'd venture to say that only you and I really know that NetZ/PIA are different), so labeling it as `NEX` for what most of our devs consider "NEX" packets, and then using separate `NetZ` and `PIA` labels would probably be more clear/helpful

Edit: I've set it to NetZ for now with a comment explaining the reasoning, we can continue to discuss it though it you still think this is an issue

DaniElectra replied:
My point was that labeling the QRV part as just "NEX" while not including it on NetZ would increase the misconception about NEX being only about the server stuff, that's why I proposed mentioning NEX alongside the specific technology "QRV" / "NetZ" alongside it.

In any case not a big deal, this works too

DaniElectra commented on src/npln/npln-transaction.ts:
```diff
@@ -0,0 +1,358 @@
+		return {
+			id: -1, // * Gets set later when emitted
+			elapsed_time: 0, // TODO - Add this
+			transport: 'NPLN', // TODO - p2p packets should change this
```
```suggestion
			transport: 'NPLN',
```

jonbarrow committed:
chore: remove outdated TODO comments

jonbarrow committed:
feat: move stack trace back to end of packet details section

jonbarrow committed:
feat: add isNetZ helper on PRUDPPacket class

jonbarrow committed:
chore: simplify NetZ NAT packet data serialization

since the NAT data fields are also in NAT stream type packets, we can skip the extra checks

jonbarrow committed:
feat: make HokakuCTR raw RMC packets its own transport

while technically still NEX, HokakuCTR is different enough to warrant its own separation imo

jonbarrow committed:
feat: add source address to NPLN packets

jonbarrow committed:
chore: rename "Headers" and "Parameters" in NPLN to "HTTP Headers" and "gRPC Parameters"

jonbarrow committed:
feat: add full type name to NPLN display types

jonbarrow committed:
feat: make NPLN use normal protobuf type names for display purposes

this helps make the UI line up with the real protobufs

jonbarrow committed:
fix: replace deprecated slice with subarray

jonbarrow committed:
fix: make NPLN maps render serialize correctly

jonbarrow committed:
feat: mark NetZ packets as NetZ

jonbarrow committed:
fix: move Session to be global

this prevents an issue where a session with thousands of packets continues to parse and emit events when a new session is opened. however it still does not clear the packet list right away, unsure why

jonbarrow committed:
chore: merge Session "packet" and "nplnTransaction" events into a single "serializedMessage" event

jonbarrow committed:
chore: use packet.isStreamTypeNAT() in Session.proccessPacket

jonbarrow committed:
fix: fix serialized message IDs always being -1 when sent to the renderer

jonbarrow committed:
feat: add PIA stub

jonbarrow committed:
chore: move packet ID handling to session rather than menu.ts

jonbarrow committed:
feat: add virtual scrolling to packet list

this helps reduce slowdowns on dumps with thousands of packets

jonbarrow committed:
feat: add source/destination addresses to PIA packets

jonbarrow committed:
chore: update electron to 40.6.1

jonbarrow committed:
fix: render RMC parents in serialized view

jonbarrow committed:
chore: rename isRMCList and isRMCMap to isList and isMap in serialized view

jonbarrow committed:
chore: remove unused css classes

jonbarrow committed:
fix: fix packet details overview section not using column count correctly

jonbarrow committed:
fix: fix NEX parent type names from being overwritten by children

calls to "this" inside parents still access the child class instance. hard-coding the names is a bit messy, but theres not really a "better" way without digging into the class prototypes, which is far uglier in my opinion

jonbarrow committed:
fix: add missing __parent to all NEX types that extend Data

jonbarrow committed:
chore: rename Structure to DDLClass and structureVersion to revision

jonbarrow committed:
feat: make all serializable fields copyable

jonbarrow committed:
chore: quit app when terminal process exits

jonbarrow committed:
feat: add visual feedback for when a value is copied in the UI

jonbarrow committed:
chore: add NetZ and PIA to support list in README

jonbarrow committed:
chore: replace preload/index.d.ts with global.d.ts

the preload file wasnt being used anyway and just contained the same types as the global file

jonbarrow committed:
feat: add settings menu in the UI

unfinished, only lets you view/edit existing NEX accounts for now. but this adds the bones

AntogamerYT committed:
feat(proxy): add NPLN proxy server

AntogamerYT committed:
feat(settings): add proxy port configuration

AntogamerYT committed:
build: add mockttp dependency

AntogamerYT committed:
feat(ui): add NPLN proxy settings page

AntogamerYT committed:
feat(ui): support live packet updates and auto-scroll

AntogamerYT committed:
feat(npln): add mockttp request/response parsing methods

AntogamerYT committed:
feat(npln): add documentation for the built-in gRPC/NPLN proxy

AntogamerYT committed:
docs: update NPLN proxy guide, add ban warning

AntogamerYT committed:
refactor(ui): simplify updatePacket logic, lint fixes

AntogamerYT committed:
style: lint fixes across UI and settings

AntogamerYT committed:
style: lint fixes

AntogamerYT committed:
feat: add custom .pnsj file format and exporting

AntogamerYT committed:
fix: remove whitespace from destinationAddress

AntogamerYT committed:
fix: check if gameServerID is not empty

jonbarrow committed:
Merge pull request #42 from AntogamerYT/feat/npln-proxy

jonbarrow committed:
chore: clean up various style nits from #42

this just cleans up some of my nits from the PR that werent worth holding the changes up for

jonbarrow committed:
chore: add "node:" prefix to zlib import in PNSJSession

jonbarrow committed:
chore: move settings button in UI to be the last in the list

jonbarrow committed:
feat: add import button to UI

jonbarrow committed:
fix: fix IconButton import path in App.vue

jonbarrow committed:
feat: add PIA overview section

data is still not parsed

jonbarrow committed:
fix: fix TypeScript build errors

jonbarrow committed:
feat: add session file drag and drop

jonbarrow committed:
feat: rebrand to Pretendo Network Viewer

jonbarrow committed:
chore: remove arrows from NPLN port settings styles and set the font to mono

jonbarrow committed:
feat: send all serialized messages at once

sending them one at a time created massive IPC overhead. the parsers can parse dumps in under a second, but would take hours to pipe all the messages to the UI. this sends them all at once which makes the UI immediately usable with all messages

jonbarrow committed:
feat: add message ID to UI

jonbarrow committed:
feat: add dump loading state indicator

jonbarrow committed:
feat: add feedback notification for when settings are saved

i tried to make this generic so we can use it for more things in the future if we want

jonbarrow committed:
feat: debounce packet searches

jonbarrow committed:
chore: pre-bake packet search strings to speed up search

old search was calling JSON.stringify for every packet on every search, which was incredibly slow. this pre-bakes the strings into a map as they get added to the UI, which speeds things up massively

jonbarrow committed:
feat: scroll to selected message in UI when filtering

jonbarrow committed:
feat: add support for reading pcapng.gz dumps

jonbarrow committed:
fix: correct PRUDPPacket RMC error name serialization

jonbarrow committed:
chore: make status badge fit better in table and shrink ID col

jonbarrow committed:
chore: give RMC error status badges the red color

jonbarrow committed:
fix: correct backwards RVConnectionData revision number check

DaniElectra reviewed (dismissed):
Would like to get feature parity with the old version (such as implementing the legacy ranking or some game-specific protocols), but won't make that a blocker for merging. LGTM

jonbarrow commented:
> Would like to get feature parity with the old version (such as implementing the legacy ranking or some game-specific protocols), but won't make that a blocker for merging. LGTM

@DaniElectra I just added every protocol, patches included, which is marked as "provided by Nintendo" on the Wikis. I didn't add EVERY protocol, since not all of them are provided by Nintendo, so we'd need to add those ones later. But this brings us up to feature parity with the original viewer plus more

DaniElectra commented on README.md:
```diff
@@ -1,70 +1,210 @@
+- [ ] [Fiddler (`.saz`)](https://docs.telerik.com/fiddler-everywhere/knowledge-base/fiddler-archives)
+- [x] [Proxide (`.bin`)](https://github.com/Rantanen/proxide)
+
+### NEX Protocols
```
The list needs updating with all the protocols supported

jonbarrow replied:
Done, I also added the `Screening` protocol because I forgot that one last time. Should be all good now

DaniElectra reviewed (approved):
LGTM! Reminder to update the repo name once merged
